pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -147,33 +146,81 @@ func CollectResourceRequests(vms []*k6tv1.VirtualMachine) []operatormetrics.Coll
 	var results []operatormetrics.CollectorResult
 
 	for _, vm := range vms {
-		memoryRequested, err := getMemoryRequested(vm)
-		if err != nil {
-			log.Log.V(4).Infof("Error getting memory request for VM %s: %v", vm.Name, err)
-			continue
-		}
-
-		results = append(results, operatormetrics.CollectorResult{
-			Metric: vmResourceRequests,
-			Value:  float64(memoryRequested),
-			Labels: []string{vm.Name, vm.Namespace, "memory", "bytes"},
-		})
+		results = append(results, collectMemoryResourceRequestsFromDomain(vm)...)
+		results = append(results, collectCpuResourceRequestsFromDomain(vm)...)
+		results = append(results, collectCpuResourceRequestsFromResources(vm)...)
 	}
 
 	return results
 }
 
-func getMemoryRequested(vm *k6tv1.VirtualMachine) (int64, error) {
+func collectMemoryResourceRequestsFromDomain(vm *k6tv1.VirtualMachine) []operatormetrics.CollectorResult {
 	if vm.Spec.Template == nil {
-		return 0, fmt.Errorf("domain spec not set")
+		return []operatormetrics.CollectorResult{}
 	}
 
 	memoryRequested := vm.Spec.Template.Spec.Domain.Resources.Requests.Memory()
 	if memoryRequested.IsZero() {
-		return 0, fmt.Errorf("memory request not set")
+		return []operatormetrics.CollectorResult{}
+	}
+
+	return []operatormetrics.CollectorResult{{
+		Metric: vmResourceRequests,
+		Value:  float64(memoryRequested.Value()),
+		Labels: []string{vm.Name, vm.Namespace, "memory", "bytes"},
+	}}
+}
+
+func collectCpuResourceRequestsFromDomain(vm *k6tv1.VirtualMachine) []operatormetrics.CollectorResult {
+	var cr []operatormetrics.CollectorResult
+
+	if vm.Spec.Template.Spec.Domain.CPU == nil {
+		return cr
+	}
+
+	if vm.Spec.Template.Spec.Domain.CPU.Cores != 0 {
+		cr = append(cr, operatormetrics.CollectorResult{
+			Metric: vmResourceRequests,
+			Value:  float64(vm.Spec.Template.Spec.Domain.CPU.Cores),
+			Labels: []string{vm.Name, vm.Namespace, "cpu", "cores"},
+		})
+	}
+
+	if vm.Spec.Template.Spec.Domain.CPU.Threads != 0 {
```

Comment:
@enp0s3 do you know if sockets and threads have an impact in the k8s use case?
I tend to think we should only add CPU cores metric like its done for pods in kube_pod_resource_request metrics.